repeater/src/main/scala/com.snowplowanalytics.snowplow.storage.bigquery.repeater/Recover.scala

```diff
@@ -0,0 +1,62 @@
+package com.snowplowanalytics.snowplow.storage.bigquery.repeater
+
+import fs2.Stream
+import cats.effect.Concurrent
+import cats.effect.Timer
+import blobstore.Path
+import blobstore.implicits.GetOps
+import io.circe.JsonObject
+import java.time.Instant
+import java.util.UUID
+import io.circe.generic.auto._, io.circe.parser._
+import cats.effect.Sync
+import cats.syntax.all._
+
+object Recover {
+
+  def recoverFailedInserts[F[_]: Timer: Concurrent](resources: Resources[F]): Stream[F, Unit] =
+    resources
+      .store
+      .list(resources.bucket)
+      .evalMap(resources.store.getContents)
+      .map(recover)
+      .evalMap {
+        case Left(e)     => Sync[F].pure(println(s"Error: $e")) // TODO: use logger
+        case Right(read) => resources.pubSubProducer.produce(read) *> Sync[F].unit
+      }
+
+  case class SimpleEventContainer(eventId: UUID, etlTstamp: Instant, payload: String)
+
+  // TODO: filter only time period that is needed (based on names in GCS bucket)
+  // TODO: filter only failures that are due to invalid column
+  // TODO: count successfuly recovered events (?)
+  def recover: String => Either[String, EventContainer] =
+    b =>
+      stringToFailedInsertBadRow(b).map { ev =>
+        val fixed = fix(ev.payload)
```

Comment:
This can potentially corrupt a lot of good data - if there are any vaules with `_%` it wil change them without a reason - it should operate only on keys.